File: androiddevkit/src/main/java/com/ams/androiddevkit/utils/connectivity/receiver/NetworkChangeReceiver.java
//package com.ams.androiddevkit.utils.connectivity.receiver;
//
//import android.content.BroadcastReceiver;
//import android.content.Context;
//import android.content.Intent;
//
//import com.ams.androiddevkit.utils.connectivity.ConnectivityService;
//import com.ams.androiddevkit.utils.services.logging.LoggingService;
//
////  INSIDE APPLICATION APP CLASS ADD THIS
////  NetworkChangeReceiver networkChangeReceiver = new NetworkChangeReceiver();
////  registerReceiver(networkChangeReceiver, new IntentFilter("android.net.conn.CONNECTIVITY_CHANGE"));
////  AND IN BASE PRESENTERS
////  addDisposable(RxBusMessages.subscribeToConnectivityChange(lifeCycle(), this,getLoggingService()));
//public class NetworkChangeReceiver extends BroadcastReceiver {
//
//    private final ConnectivityService connectivityService;
//    private final LoggingService loggingService;
//
//    public NetworkChangeReceiver(ConnectivityService connectivityService, LoggingService loggingService) {
//        super();
//        this.connectivityService = connectivityService;
//        this.loggingService = loggingService;
//        BootstrapRxConnectivityBus.getRxBus(connectivityService).send(new BootstrapIsOnline(connectivityService.haveActiveNetwork()));
//    }
//
//    @Override
//    public void onReceive(final Context context, final Intent intent) {
//        loggingService.v("NetworkChangeReceiver", "OnReceive: haveActiveNetwork:" + connectivityService.haveActiveNetwork());
//        BootstrapRxConnectivityBus.getRxBus(connectivityService).send(new BootstrapIsOnline(connectivityService.haveActiveNetwork()));
//    }
//}

File: androiddevkit/src/main/java/com/ams/androiddevkit/utils/internetConnectionManager/NetworkChangeReceiver.kt
package com.ams.androiddevkit.utils.internetConnectionManager

import android.content.BroadcastReceiver
import android.content.Context
import android.content.Intent
import android.net.ConnectivityManager
import android.os.Build

open class NetworkChangeReceiver: BroadcastReceiver() {

    protected open val connectivityListener: IConnectivityListener by lazy {
        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.N) {
            NewConnectivityListener()
        }
        else {
            LegacyConnectivityListener()
        }
    }

    override fun onReceive(context: Context, intent: Intent) {
        val connectivityManager = context.getSystemService(Context.CONNECTIVITY_SERVICE) as ConnectivityManager
        connectivityManager.let { connectivityListener.listenForNetworkChanges(it) }
    }
}
